source_code/main_mcu/src/EMU/dataflash.c

```diff
@@ -11,7 +11,7 @@ void emu_dataflash_init(const char *path)
     int i;
     const char *bundle_paths[] = {
     #ifndef WIN32    
-        "/usr/share/misc/miniblebundle.img",
+        XSTR(DESTDIR) XSTR(PREFIX) "/share/misc/miniblebundle.img",
```

Comment:
@saagit we're currently having an issue where, when built locally, the emulator is looking for 
"DESTDIRPREFIX/share/misc...." 
I'm therefore guessing that as these 2 vars are not defined the XSTR() macro is interpreting DESTDIR as "DESTDIR" ?
Could you help us on this one?

Reply:
Or perhaps it is OS-related, as this https://onlinegdb.com/rkkLRdtKL seems to work?
Perhaps we should check in the makefile if the defines are absent and then set to "" ?

Reply:
Sure, I can take a look, but it seems quite weird.  What build platform is this happening on?

Reply:
@deXol  ?

Reply:
I just took another look at it and came up with another question.  Is source_code/main_mcu/Makefile.emu being used to build?  The way this works is that Makefile.emu passes in its values of DESTDIR and PREFIX to the C compiler via this line:
`C_DEFINES += -DDESTDIR=$(DESTDIR) -DPREFIX=$(PREFIX)`
If they are unset, they are passed in as empty strings already.  The problem appears to be that somehow the C compiler is getting invoked without DESTDIR and PREFIX being set.  As an example, when I do `make -f Makefile.emu` I see it invoking gcc every time with a bunch of arguments.  Here's an example one:
`Building file: Release-emu/src/FILESYSTEM/custom_bitstream.o
Invoking: GNU C Compiler
gcc -DNDEBUG -Os -fdata-sections -ffunction-sections -Wall -c -pipe -fno-strict-aliasing -Werror-implicit-function-declaration -Wpointer-arith -ffunction-sections -fdata-sections -Wchar-subscripts -Wcomment -Wformat=2 -Wmain -Wparentheses -Wsequence-point -Wreturn-type -Wswitch -Wtrigraphs -Wunused -Wuninitialized -Wunknown-pragmas -Wundef -Wshadow -Wwrite-strings -Wsign-compare -Wmissing-declarations -Wformat -Wmissing-format-attribute -Wno-deprecated-declarations -Wpacked -Wredundant-decls -Wunreachable-code -Wcast-align -Wlogical-op -fPIC -Wstrict-prototypes -Wmissing-prototypes -Wimplicit-int -Wbad-function-cast -Wnested-externs -Wjump-misses-init -Wlong-long -Wfloat-equal -Waggregate-return -std=gnu99 -DBR_BE_UNALIGNED=0 -DBR_CT_MUL15=0 -DBR_ENABLE_INTRINSICS=0 -DBR_CT_MUL31=0 -DBR_LE_UNALIGNED=0 -DBR_NO_ARITH_SHIFT=0 -DBR_POWER_ASM_MACROS=0 -D_ARCH_PWR8=0 -DBR_POWER8=0 -DEMULATOR_BUILD -DDESTDIR= -DPREFIX=/usr -I"src/EMU" -I"src" -I"src/config" -I"src/PLATFORM" -I"src/CLOCKS" -I"src/SERCOM" -I"src/FLASH" -I"src/FILESYSTEM" -I"src/DMA" -I"src/TIMER" -I"src/SMARTCARD" -I"src/OLED" -I"src/ACCELEROMETER" -I"src/INPUTS" -I"src/COMMS" -I"src/LOGIC" -I"src/SECURITY" -I"src/GUI" -I"src/NODEMGMT" -I"src/RNG" -I"src/BearSSL/src" -I"src/BearSSL/inc"  -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I/usr/include/x86_64-linux-gnu/qt5/QtWidgets -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtGui -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtNetwork -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I/usr/include/x86_64-linux-gnu/qt5 -MD -MP -MF "Release-emu/src/FILESYSTEM/custom_bitstream.d" -MT "Release-emu/src/FILESYSTEM/custom_bitstream.o" -o "Release-emu/src/FILESYSTEM/custom_bitstream.o" "src/FILESYSTEM/custom_bitstream.c"
Finished building: Release-emu/src/FILESYSTEM/custom_bitstream.o
`
By looking carefully, you can find `-DDESTDIR= -DPREFIX=/usr` buried in that output.  I double-checked Makefile.emu and none of this appears to be in platform specific code.

In summary, if Makefile.emu is not being used, that would be the problem.  If Makefile.emu is being used, I need more information and logging to understand how this can be happening.

Reply:
building with using the Qt pro file on Ubuntu 19.10

Reply:
Sorry I commented instead of replying last time, but the following resolves the problem in a hardcoded way:

```
diff --git a/source_code/main_mcu/minible_emu.pro b/source_code/main_mcu/minible_emu.pro
index af69ee0d..3db9a30b 100644
--- a/source_code/main_mcu/minible_emu.pro
+++ b/source_code/main_mcu/minible_emu.pro
@@ -144,6 +144,8 @@ QMAKE_CXXFLAGS += -fdata-sections \
     -fPIC
     
 DEFINES += EMULATOR_BUILD
+DEFINES += DESTDIR=""
+DEFINES += PREFIX="/usr"
 
 HEADERS  += src/MainWindow.h \ \
     src/BearSSL/inc/bearssl.h \

```

I'm now going to look into qtcreator's idea of installing and its built in DESTDIR variable to see if I can get that hooked together with the compiler.  Regardless, a pull request should be coming in before the end-of-day to fix this.

Reply:
Ok, [pull request](https://github.com/mooltipass/minible/pull/119) made to fix.  Thanks for letting me know.